libflux/flux-core/src/semantic/formatter/tests.rs

```diff
@@ -0,0 +1,118 @@
+use crate::semantic::convert_source;
+use crate::semantic::formatter::format;
+use pretty_assertions::assert_eq;
+
+#[test]
+fn literals() {
+    let sem_pkg = convert_source(
+        r#"
+            a = "Hello, World!"
+            b = 12
+            c = 18.5
+            d = 12h
+            e = 2019-10-31T00:00:00Z
+            f = /server[01]/
+        "#,
+    )
+    .unwrap();
+
+    let got = format(&sem_pkg).unwrap();
+
+    let want = r#"package main
+a = "Hello, World!":string
+b = 12:int
+c = 18.5:float
+d = 043200000000000false:duration
+e = 2019-10-31T00:00:00Z:datetime
+f = /server[01]/:regex"#;
+
+    assert_eq!(want, got);
+}
+
+#[test]
+fn array_lit() {
+    let sem_pkg = convert_source(
+        r#"
+a = [1, 2, 3]
+b = [1.1, 2.2, 3.3]
+c = ["1", "2", "3"]
+d = [1s, 2m, 3h]
+e = [2019-10-31T00:00:00Z]
+f = [/a/, /b/, /c/]
+g = [{a:0, b:0.0}, {a:1, b:1.1}]
+        "#,
+    )
+    .unwrap();
+
+    let got = format(&sem_pkg).unwrap();
+
+    let want = r#"package main
+a = [1:int2:int3:int]
+b = [1.1:float2.2:float3.3:float]
+c = ["1":string"2":string"3":string]
+d = [01000000000false:duration0120000000000false:duration010800000000000false:duration]
+e = [2019-10-31T00:00:00Z:datetime]
+f = [/a/:regex/b/:regex/c/:regex]
+g = [{a: 0:int, b: 0.0:float}{a: 1:int, b: 1.1:float}]"#;
```

Comment:
These elements should have commas separating them. Can you also show the type for the array itself, like `[1:int, 2:int]:[int]`?